apimodels/agent_models.go

```diff
@@ -105,6 +106,16 @@ type CreateHost struct {
 	SetupTimeoutSecs    int    `mapstructure:"timeout_setup_secs" json:"timeout_setup_secs"`
 	TeardownTimeoutSecs int    `mapstructure:"timeout_teardown_secs" json:"timeout_teardown_secs"`
 	Retries             int    `mapstructure:"retries" json:"retries"`
+
+	// docker-related settings
+	Image    string           `mapstructure:"image" json:"image" plugin:"expand"`
+	Command  string           `mapstructure:"command" json:"command" plugin:"expand"`
+	Registry RegistrySettings `mapstructure:"registry" json:"registry" plugin:"expand"`
+
+	Background    bool   `mapstructure:"background" json:"background"`
+	PollFrequency int    `mapstructure:"poll_frequency_secs" json:"poll_frequency_secs"`
+	StdoutFile    string `mapstructure:"stdout_file_name" json:"stdout_file_name" plugin:"expand"`
```

Comment:
I can imagine an objection that people will have is that, if they have stdout and stderr logs, how can they unify these logs? If they don't have timestamps, it will be hard to know when things occurred. I don't think combining the logs fixes this problem, since then we can't reconcile the logs within the polling interval. Is there something available in docker that will send stderr logs to stdout? Are stdout and stderr logs meaningfully separated in docker?

Reply:
There is an option for container logs to have timestamps--having that is probably the most meaningful. With the timestamps set to true, we could either leave stderr and stdout separate (which is implied by the current Stdout/Stderr file params) or parse the timestamps ourselves and organize stdout/stderr into one cohesive log. Docker itself automatically separates the logs and doesn't seem to have a solution to combining them.

Reply:
I think including the timestamps in the files but not doing any processing of them ourselves makes sense.